XX.md

```diff
@@ -0,0 +1,1078 @@
+
+**Implementation Requirements:**
+
+Relays implementing partial replication MUST:
+
+1. **Use Cryptographic RNG**: Use cryptographically secure random number generation (e.g., `crypto/rand` in Go, `crypto.getRandomValues()` in JavaScript)
+2. **Per-Event Independence**: Each event must get an independent random roll
+3. **No Bias**: Random selection must be uniform and unbiased
+4. **Deterministic Recording**: Once decision is made, it must be consistent (no re-rolling)
+5. **Event Integrity**: Replicated events must be complete and unmodified
+
+**Optional Enhancements:**
+
+Relays MAY implement additional strategies:
+
+- **Priority Boosting**: Increase probability for directory events (Kinds 0, 3, 5, etc.)
+- **Kind-Specific Rates**: Apply different trust levels to different event kinds
+- **Time-Based Adjustment**: Vary trust level based on network load or time of day
+- **Reputation Weighting**: Boost probability for high-reputation users
+
+However, the base trust_level percentage MUST always be respected as the minimum probability.
+
+**Event Kind Replication:**
+- **Directory Events**: Always replicated regardless of `K` tag (kinds 0, 3, 5, 1984, 10002, 10000, 10050)
+- **Custom Kinds**: Additional event kinds specified in the `K` tag are replicated based on trust level
+- **Specialization**: Enables relay operators to specialize in specific data types (e.g., long-form content, marketplace events, etc.)
+- **Near Real-time**: Events matching `K` tag kinds are replicated with minimal delay
+- **Bidirectional**: Replication occurs both to and from the trusted relay for specified kinds
+
+### Group Tag Act (Kind 39102)
+
+Group Tag Acts establish ownership and control over arbitrary string tags, functioning as a first-come-first-served registration system akin to domain name registration. These tags form the foundation for permissioned structured groups that can span multiple relays while maintaining consistent state. They also constitute a name registration system akin to DNS, and could have IP adress routing information attached to it with events (best to be CRDT add/remove so it's idempotent).
+
+Since the uses of group identity names and webserver identity information (and it logically would include most of the same things you find in DNS records) tend to overlap, in that finding the group and going to their static content (such as structured long form documents and wikis) is the same place you want to go to read public messages and discover new friends. This allows progressive replication and redundancy for communities that enables the network to scale to larger userbases without a linear reduction in performance.
+
+**Registration Model:**
+- Group tags are **alienable** (transferable) and follow first-come-first-served registration
+- The first valid Group Tag Act for a given `group_id` establishes initial ownership
+- Ownership can be transferred through Group Tag Transfer events (Kind 39106) with optional escrow
```

Comment:
This runs up against the double spend problem, so unless you're anchoring in a block chain this isn't going to work